Kros.Utils/src/Kros.Utils/Data/BulkActions/BulkUpdateBase.cs

```diff
@@ -72,27 +70,37 @@ public void Update(IBulkActionDataReader reader)
             }
         }
 
-        /// <summary>
-        /// Updates all data from source <paramref name="reader"/>.
-        /// </summary>
-        /// <param name="reader">Data source.</param>
-        public void Update(IDataReader reader)
+        /// <inheritdoc/>
+        public async Task UpdateAsync(IBulkActionDataReader reader)
+        {
+            using (var bulkUpdateReader = new BulkActionDataReader(reader))
+            {
+                await UpdateAsync(bulkUpdateReader);
+            }
+        }
+
+        /// <inheritdoc/>
+        public void Update(IDataReader reader) =>
+            UpdateCoreAsync(reader, useAsync: false).GetAwaiter().GetResult();
+
+        /// <inheritdoc/>
+        public Task UpdateAsync(IDataReader reader) =>
+            UpdateCoreAsync(reader, useAsync: true);
+
+        private async Task UpdateCoreAsync(IDataReader reader, bool useAsync)
         {
             var tempTableName = CreateTempTable(reader);
 
             using (ConnectionHelper.OpenConnection(_connection))
             {
-                InsertIntoTempTable(reader, tempTableName);
+                await InsertIntoTempTableAsync(reader, tempTableName, useAsync);
```

Comment:
Just question: maybe `InvokeAction` and `DoneTempTable` can be async too?

Reply:
- `DoneTempTable` changed to `DoneTempTableAsync`.
- `InvokeAction` I can do it. But I'm not sure if we want to. Because there would have to be twin for `TempTableAction` -> `TempTableActionAsync` and it would have to be documented when to use it. Should I do it?